### ISSUE TITLE
Consider using DictVectorizer and semi-supervised learning to see if any generalizations arise from using a neural network. Review contrastive loss and ideas here

### ISSUE THREAD

**Issue by Shuyib** [Shuyib/Phylogenetic-tree-study#44]
The dictVectorizer will not work so well. We have variable lengths of the sequences. Therefore, [embeddings](https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html) have an argument padding in order to make the sequences of the same length.

This workflow allows use to make a representation of the data with dictionary structure that is, an embedding. Which we can use for the semi-supervised or unsupervised methods. Which we can use loss functions like contrastive loss to examine similarity and differences.

**Comment by Shuyib** [comment]
Seems this was repeated. I'll try make a naive version so that we can build upon this.
